orbit/pkg/useraction/mdm_migration_darwin.go

```diff
@@ -373,27 +373,36 @@ func (m *swiftDialogMDMMigrator) renderMigration() error {
 	return nil
 }
 
+// Show displays the dialog every time is called, as long as there isn't a
+// dialog already open.
 func (m *swiftDialogMDMMigrator) Show() error {
-	if m.showMu.TryLock() {
-		defer m.showMu.Unlock()
+	select {
+	case m.showCh <- struct{}{}:
+		defer func() { <-m.showCh }()
+	default:
+		log.Info().Msg("there's a migration dialog already open, refusing to launch")
+		return nil
+	}
 
-		if err := m.renderMigration(); err != nil {
-			return fmt.Errorf("show: %w", err)
-		}
+	if err := m.renderMigration(); err != nil {
+		return fmt.Errorf("show: %w", err)
 	}
 
+	m.lastShown = time.Now()
 	return nil
 }
 
+// ShowInterval acts as a rate limiter for Show, it only calls the function IIF
+// m.frequency has passed since the last time the dialog was successfully
+// shown.
 func (m *swiftDialogMDMMigrator) ShowInterval() error {
-	if m.intervalMu.TryLock() {
-		defer m.intervalMu.Unlock()
-		if time.Since(m.lastShown) > m.frequency {
-			if err := m.Show(); err != nil {
-				return fmt.Errorf("show interval: %w", err)
-			}
-			m.lastShown = time.Now()
-		}
+	if time.Since(m.lastShown) <= m.frequency {
```

Comment:
thank you! I was hoping to avoid the double locking, especially since `Show` is called from `ShowInterval`. Feels like a smell with the way things are currently organized.

I added the locks using `sync.RWMutex` as you suggested, it's not only fancier but helps cement the concept that `lastShown` is only being set in `Show`